### DISCUSSION TITLE
JITParticle has no attribute 'add_variables'

### DISCUSSION THREAD

**Discussion by TomMBIO** [Parcels-code/Parcels#1504] [General]
Hi there,

I'm working through the getting started with parcels tutorials (https://docs.oceanparcels.org/en/latest/examples/parcels_tutorial.html) and am having trouble getting this code chunk to work:

extra_vars = [
    Variable("distance", initial=0.0, dtype=np.float32),
    Variable("prev_lon", dtype=np.float32, to_write=False, initial=attrgetter("lon")),
    Variable("prev_lat", dtype=np.float32, to_write=False, initial=attrgetter("lat")),
]

DistParticle = JITParticle.add_variables(extra_vars)

**Here is the error message:**

__AttributeError                            Traceback (most recent call last)
Cell In[6], line 1
----> 1 DistParticle = JITParticle.add_variables(extra_vars)

AttributeError: type object 'JITParticle' has no attribute 'add_variables'__

I think the attribute it is calling for no longer exists, can you help? I would like to understand in case I encounter this in the future.

Thanks,
Tom

**Comment by TomMBIO** [comment]
Apologies, was supposed to post this in issues. Won't happen again

**Comment by erikvansebille** [comment]
Ag, what a coincidence. This is because we _just_ released the new [Parcels v3.0.2](https://github.com/OceanParcels/parcels/releases/tag/v3.0.2) a few hours ago, which has this new feature. This probably happened after you installed Parcels on your local system...

You can do two things:
1. Update your conda/pip environment to use the latest v3.0.2 (preferred)
2. Refer to the v3.0.1 version of the documentation at https://docs.oceanparcels.org/en/v3.0.1/documentation.html

**Reply by TomMBIO** [reply]
This fixed the issue, thanks for your speedy response! I will be in touch on here about future problems I face if that's okay.

KR
Tom
